core/string/ustring.cpp

```diff
@@ -4306,6 +4306,96 @@ String String::replace_first(const char *p_key, const char *p_with) const {
 	return *this;
 }
 
+String String::replace_char(char32_t p_key, char32_t p_with) const {
+	ERR_FAIL_COND_V_MSG(p_with == 0, String(), "`with` must not be null.");
```

Comment:
I think returning `*this` is more appropriate, as part of the contract that `replace_char` does not change the length.

Reply:
Good point will change